File: sdk/samples/WPFSamples/ActivationSample/csharp/maildispatcher.cs
using System;
using System.Windows.Threading;

namespace ActivationSample
{
    public class MailDispatcher
    {
        DispatcherTimer timer;

        public event EventHandler MailDispatched;

        public void Start()
        {
            this.timer = new DispatcherTimer();
            timer.Tick += timer_Tick;
            timer.IsEnabled = true;
            timer.Interval = new TimeSpan(0, 0, 5);
        }

        public void Stop()
        {
            this.timer.IsEnabled = false;
            this.timer.Tick -= timer_Tick;
            this.timer = null;
        }

        void timer_Tick(object sender, EventArgs e)
        {
            if (MailDispatched != null) MailDispatched(this, EventArgs.Empty);
        }
    }
}

File: sdk/samples/WPFSamples/ActivationSample/visualbasic/maildispatcher.vb
Imports System.Windows.Threading

Public Class MailDispatcher

    Private timer As DispatcherTimer

    Public Event MailDispatched As EventHandler

    Public Sub Start()
        Me.timer = New DispatcherTimer
        AddHandler Me.timer.Tick, New EventHandler(AddressOf Me.timer_Tick)
        Me.timer.IsEnabled = True
        Me.timer.Interval = New TimeSpan(0, 0, 5)
    End Sub

    Public Sub [Stop]()
        Me.timer.IsEnabled = False
        RemoveHandler Me.timer.Tick, New EventHandler(AddressOf Me.timer_Tick)
        Me.timer = Nothing
    End Sub


    Private Sub timer_Tick(ByVal sender As Object, ByVal e As EventArgs)
        RaiseEvent MailDispatched(Me, EventArgs.Empty)
    End Sub

End Class

